GNUmakefile

```diff
@@ -5,16 +5,17 @@ HOSTNAME     = registry.terraform.io
 NAMESPACE    = dnsimple
 BINARY       = terraform-provider-${PKG_NAME}
 VERSION      = $(shell git describe --tags --always | cut -c 2-)
-OS_ARCH      = darwin_$(shell uname -m)
+OS_ARCH      := $(shell echo "$$(uname -s)_$$(go env GOARCH)" | tr A-Z a-z)
 
 default: build
 
 build: fmtcheck
 	go install
 
 install: build
-	mkdir -p ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${PKG_NAME}/${VERSION}/${OS_ARCH}
-	mv ${GOPATH}/bin/${BINARY} ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${PKG_NAME}/${VERSION}/${OS_ARCH}
+# VERSION contains the Git commit, which is not a valid version for Terraform. We also use 0.0.1 so that the version never conflicts with versions from the registry, and also so it's easy to see when a local override is being used.
+	mkdir -p ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${PKG_NAME}/0.0.1/${OS_ARCH}
+	mv ${GOPATH}/bin/${BINARY} ~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${PKG_NAME}/0.0.1/${OS_ARCH}
```

Comment:
I am more in favor of leaving it as it was. As any developer can set the **VERSION** when running the make command to fit their use case. i.e. `make install VERSION=1.0.1`